src/PortingAssistantExtensionClientShared/Commands/SolutionAssessmentCommand.cs

```diff
@@ -100,15 +100,16 @@ private async void Execute(object sender, EventArgs e)
             {
                 if (!await CommandsCommon.CheckLanguageServerStatusAsync()) return;
                 if (!CommandsCommon.SetupPage()) return;
-                CommandsCommon.EnableAllCommand(false);
-                var SolutionFile = await CommandsCommon.GetSolutionPathAsync();
-                SolutionName = Path.GetFileName(SolutionFile);
+                // Verify Target framework selection before disabling all commands.
                 if (UserSettings.Instance.TargetFramework.Equals(TargetFrameworkType.NO_SELECTION))
                 {
                     if (!SelectTargetDialog.EnsureExecute()) return;
                 }
+                CommandsCommon.EnableAllCommand(false);
+                var solutionFile = await CommandsCommon.GetSolutionPathAsync();
+                SolutionName = Path.GetFileName(solutionFile);
                 string pipeName = Guid.NewGuid().ToString();
-                CommandsCommon.RunAssessmentAsync(SolutionFile, pipeName);
+                await CommandsCommon.RunAssessmentAsync(solutionFile, pipeName);
```

Comment:
Please remove the await unless there's a reason to add here. This code invokes the command and waits for a response using the Pipe initiated below it. Adding an await will cause a timeout for larger solutions.